### PR TITLE
feat(sdk): support compiling components with `dsl.PipelineTaskFinalStatus` type

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Supports the following:

```python
from kfp import dsl
from kfp import compiler
from kfp import components


@dsl.component
def exit_comp(status: dsl.PipelineTaskFinalStatus):
    print(status)


package_path = __file__.replace('.py', '.yaml')
compiler.Compiler().compile(pipeline_func=exit_comp, package_path=package_path)
loaded_exit_comp = components.load_component_from_file(package_path)
```

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this 
repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/test_data/components/component_with_task_final_status.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+              componentInputParameter: status
+        taskInfo:
+          name: exit-comp
+  inputDefinitions:
+    parameters:
+      status:
+        isOptional: true
+        parameterType: TASK_FINAL_STATUS
+schemaVersion: 2.1.0
+sdkVersion: kfp-2.0.0-beta.13
```
Comment (chensun) [inline sdk/python/test_data/components/component_with_task_final_status.yaml:64]: Can you recompile to avoid referencing the polluted version?

**Review by chensun — COMMENTED** [review]
/lgtm

/hold in a favor of merging other PRs which are in progress of testing and merging first

File: sdk/python/test_data/components/component_with_task_final_status.yaml (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/python/test_data/components/component_with_task_final_status.yaml]: Good catch. Updated.

**Review by chensun — APPROVED** [review]
/lgtm
/approve
/unhold

**Pull request merged**